apigw-cache-invalidation-eventbridge/template.yaml

```diff
@@ -0,0 +1,306 @@
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - events.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: AllowAPIGatewayAccess
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "execute-api:InvalidateCache"
+                  - "execute-api:Invoke"
+                Resource: !Sub
+                  - "arn:aws:execute-api:${AWS::Region}:${AWS::AccountId}:${APIId}/Prod/GET/pets/*"
+                  - APIId: !Ref Api
+
+  CacheInvalidationRule:
+    Type: AWS::Events::Rule
+    Properties:
+      Description: "Cache Invalidation Rule"
+      EventBusName: !GetAtt PetEventBus.Name
+      EventPattern:
+        source:
+          - com.acme.pet
+        detail-type:
+          - PetUpdated
+      Targets:
+        - Id: APIGatewayTarget
+          Arn: !Sub
+            - "arn:aws:execute-api:${AWS::Region}:${AWS::AccountId}:${APIId}/Prod/GET/pets/*"
+            - APIId: !Ref Api
+          RoleArn: !GetAtt CacheInvalidationTargetRole.Arn
+          HttpParameters:
+            HeaderParameters:
+              Cache-Control: max-age=0
+            PathParameterValues:
+              - $.detail.data.id
+          DeadLetterConfig:
+            Arn: !GetAtt CacheInvalidationTargetDeadLetterQueue.Arn
+
+  CacheInvalidationTargetDeadLetterQueue:
+    Type: AWS::SQS::Queue
+
+  Api:
+    Type: AWS::Serverless::Api
+    Properties:
+      StageName: Prod
+      CacheClusterEnabled: true
+      CacheClusterSize: "0.5"
+      MethodSettings:
+        - HttpMethod: GET
+          CachingEnabled: true
+          CacheTtlInSeconds: 3600 # 1 hour
+          ResourcePath: "/pets/{id}"
+          LoggingLevel: INFO
+          DataTraceEnabled: true
+      DefinitionBody:
+        "Fn::Transform":
+          Name: "AWS::Include"
+          Parameters:
+            Location: "./api.yaml"
+
+  APICacheInvalidationDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Condition: deployMetricsDashboard
+    Properties:
+      DashboardBody: !Sub
+        - |
+          {
+            "widgets": [
+                {
+                    "height": 6,
+                    "width": 6,
+                    "y": 0,
+                    "x": 12,
+                    "type": "metric",
+                    "properties": {
+                        "metrics": [
+                            [ "AWS/ApiGateway", "CacheMissCount", "ApiName", "DynamoDB Crud API", { "region": "${AWS::Region}" } ],
+                            [ ".", "CacheHitCount", ".", ".", { "region": "${AWS::Region}" } ]
+                        ],
+                        "view": "timeSeries",
+                        "stacked": false,
+                        "region": "${AWS::Region}",
+                        "period": 30,
+                        "stat": "Average",
```

Comment:
Suggestion: Could you use stat as Sum for Cache Hit & Miss Count? Sum would provide actual value of requests that Hit or Missed the cache.